include/openmc/weight_windows.h

```diff
@@ -0,0 +1,81 @@
+#ifndef OPENMC_WEIGHT_WINDOWS_H
+#define OPENMC_WEIGHT_WINDOWS_H 1
+
+#include "openmc/mesh.h"
+#include "openmc/particle.h"
+
+#define WEIGHT_CUTOFF 1.0E-38 // default low weight cutoff
+
+namespace openmc {
+
+void read_weight_window_xml(); //!< read the weight window file
+
+namespace weight_window {
+
+// generic struct to contain the ww settings
+struct WWParams {
+  double upper_ratio;
+  double multiplier;
+  double survival_ratio;
+  int max_split;
+  double weight_cutoff;
+  std::vector<double> energy_bounds;
+  std::vector<double> lower_ww;
+  std::vector<double> upper_ww;
+};
+
+// struct to pass specific location data 
+struct ParticleWeightParams {
+  // 0,1,0.5,1,1e-12
+  ParticleWeightParams() : lower_weight(0), upper_weight(1), survival_weight(0.5), max_split(1), weight_cutoff(WEIGHT_CUTOFF) {}
```

Comment:
Default values like these can be added as initializers of the data members below. Then we can rely on the default constructor for `ParticleWeightParams`.